### ISSUE TITLE
SocketIO Connection Problems (Version 4.2.0(NodeJS) and Version 15.2.0)

### ISSUE THREAD

**Issue by Pic2490** [socketio/socket.io-client-swift#1378]
I'm having connection problems

Xcode Log:

> 2021-09-04 13:11:08.455689-0700 V.1[85386:1441059] ERROR SocketEnginePolling: Error during long poll request
2021-09-04 13:11:08.455842-0700 V.1[85386:1441059] ERROR SocketEngine: Error
2021-09-04 13:11:08.455981-0700 V.1[85386:1440613] ERROR SocketManager: Error
2021-09-04 13:11:08.456167-0700 V.1[85386:1440613] LOG SocketIOClient{/}: Handling event: error with data: ["Error"]

SwiftUI Code:

```
.onAppear{
    let socket = self.manager.defaultSocket
    socket.on(clientEvent: .connect) {data, ack in
          print("socket connected")
      }
            
      socket.connect()
}
```

NodeJS Code:

```

const http = require('http').createServer(app);
const io = require('socket.io')(http);


io.on('connection', (socket) => {
    
  socket.on('room', (room) =>{
    console.log(chalk.greenBright("Someone Joined"))
    socket.join(room)
  })
  
  
  socket.on('chat message', (msg) => {
      io.emit('chat message', msg);
    });
});


app.set('port', process.env.PORT || 3000);
http.listen(app.get('port'), () => {
    console.log(chalk.green('Listening on port 3000'));
});
```

I saw a similar issue opened a couple months ago with no resolution:

[Link To Other Open Issue](https://github.com/socketio/socket.io-client-swift/issues/1355)

Would it be best if I just downgraded versions if there's no current fix? Thanks

**Comment by Pic2490** [comment]
Downgraded SocketIO version to 2.4.0 in NodeJS just now and it worked for me

**Comment by Pic2490** [comment]
Implemented changes from compatibility table and it worked fine: 

https://github.com/socketio/socket.io-client-swift/blob/master/Usage%20Docs/Compatibility.md

**Comment by CT77777** [comment]
Thanks for offer the compatibility table.
I fix it by using Version 4.2.6(NodeJS) and Version 15.x (Swift)
